cmd/contour/serve.go

```diff
@@ -161,235 +163,202 @@ func registerServe(app *kingpin.Application) (*kingpin.CmdClause, *serveContext)
 	return serve, ctx
 }
 
-// doServe runs the contour serve subcommand.
-func doServe(log logrus.FieldLogger, ctx *serveContext) error {
+type Server struct {
```

Comment:
+1 for pulling this out into an object so we can refactor into its own package potentially at some point for test coverage etc.